File: OPENAPI-REST-API/openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/MailAlertUpdateRequest.scala

package org.openapitools.client.model


case class MailAlertUpdateRequest (
    /* Alert ID to update. */
    _alert_id: Option[Integer],
    /* Alert type identifier. */
    _type: Option[String],
    /* Alert value or threshold. */
    _value: Option[String],
    /* Email address to notify. */
    _to: Option[String],
    /* Whether the alert is enabled. */
    _enabled: Option[String]
)
object MailAlertUpdateRequest {
    def toStringBody(var_alert_id: Object, var_type: Object, var_value: Object, var_to: Object, var_enabled: Object) =
        s"""
        | {
        | "alert_id":$var_alert_id,"type":$var_type,"value":$var_value,"to":$var_to,"enabled":$var_enabled
        | }
        """.stripMargin
}

File: OPENAPI-REST-API/openapi-client/zapier/models/MailAlertUpdateRequest.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}alert_id`,
                label: `Alert ID to update. - [${labelPrefix}alert_id]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}type`,
                label: `Alert type identifier. - [${labelPrefix}type]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}value`,
                label: `Alert value or threshold. - [${labelPrefix}value]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}to`,
                label: `Email address to notify. - [${labelPrefix}to]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}enabled`,
                label: `Whether the alert is enabled. - [${labelPrefix}enabled]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'alert_id': bundle.inputData?.[`${keyPrefix}alert_id`],
            'type': bundle.inputData?.[`${keyPrefix}type`],
            'value': bundle.inputData?.[`${keyPrefix}value`],
            'to': bundle.inputData?.[`${keyPrefix}to`],
            'enabled': bundle.inputData?.[`${keyPrefix}enabled`],
        }
    },
}
